Site Switcher: Update 'Add new site' button styles

danielbachhuber opened the pull request:
Fixes #67041

Originally #67346

## Proposed Changes

Updates the styling for the 'Add new site' button in the Site Switcher.

Uses the `primary` button style.

### Calypso

<img width="483" src="https://user-images.githubusercontent.com/1500769/188759640-b992cbcb-493c-41d0-a68d-014310b0bf90.png">

### Jetpack Cloud

<img width="452" src="https://user-images.githubusercontent.com/1500769/188759394-5e9a95b4-7d94-4c14-b709-946371e8464a.png">

## Testing Instructions

1. Run Calypso locally (`yarn start`).
2. Verify the 'Add new site' button appears as expected with a variety of Dashboard Color Schemes.
3. Run Jetpack Cloud locally (`yarn start-jetpack-cloud-p`).
4. Verify the 'Add new site' button appears as expected.

danielbachhuber committed:
Site Switcher: Update "Add New Site" button styles

danielbachhuber commented:
Need to do some more work on the color schemes

danielbachhuber committed:
Add sidebar button styles for first row of dashboard color schemes

danielbachhuber committed:
Add sidebar button styles for the remaining dashboard color schemes

danielbachhuber committed:
Better hover styling for Jetpack

p-jackson commented on client/components/site-selector/add-site.tsx:
```diff
@@ -19,15 +19,13 @@ const SiteSelectorAddSite: FunctionComponent = () => {
-			</Button>
-		</span>
+		<Button
+			className="site-selector__add-new-site"
```
@danielbachhuber I didn't think we'd need to define all the custom colours for this button, because from the mockups I got the impression it was just supposed to be a "primary" button.

```suggestion
			primary
```

![Sep-07-2022 09-45-52](https://user-images.githubusercontent.com/1500769/188746948-08136924-e1ef-4876-a9eb-42a5018f994c.gif)


There are some themes where the hover colour for primary buttons are a little odd.

![Sep-07-2022 09-45-48](https://user-images.githubusercontent.com/1500769/188746714-5812d0ce-4a61-46f9-a201-ab7c02e38241.gif)

But the users will be used to it, since it's the same primary button hover used everywhere.

danielbachhuber replied:
> There are some themes where the hover colour for primary buttons are a little odd.

@p-jackson I think I like my styles better...

p-jackson replied:
Well they are better ❤️
Fair enough, I'm fine with the custom colours. The inverted colour scheme of the sidebar justifies the deviation from the norm I think.

p-jackson commented on client/jetpack-cloud/style.scss:
```diff
@@ -320,6 +320,20 @@ html {
+	}
+	.site-selector__add-new-site {
+		border-color: var( --color-sidebar-text );
+		border-radius: 2px;
```
<img width="401" alt="CleanShot 2022-09-07 at 09 50 26@2x" src="https://user-images.githubusercontent.com/1500769/188747260-58ef9957-d8a6-4361-a61f-f88db1f5a5af.png">

Looks like Jetpack Cloud has it's own border radius value, so this rule ends up just getting ignored.

p-jackson committed:
Use standard primary button style for "Add new sites" button

p-jackson committed:
Pin new site button to the bottom of the site switcher in JP Cloud

Removing the `display: block` rule was enough to move it to the bottom
(previously it was a cloud-only override to prevent it from being at the
bottom like Calypso). However there was a mysterious extra margin below
the button. I couldn't figure out where this was coming from.

The main JP cloud sidebar uses `margin-top: auto` to pin the `wp-admin`
link to the bottom, so I did the same in the site switcher. That was
enough to remove the margin.

p-jackson reviewed (commented):
Pairing with Daniel, I've updated the buttons to use the `primary` style. The custom hover style Daniel created often looked better, but the intention from the mockup appears to be that the "add new site" button now uses the bright accent colour.

I've also pinned the button to the bottom of the sidebar in JP Cloud. 💯 points for consistency.

danielbachhuber commented:
Thanks for working with me on this, @p-jackson !

zaguiini commented on client/jetpack-cloud/style.scss:
```diff
@@ -318,8 +318,8 @@ html {
 }
 
-.site-selector {
-	display: block;
```
@danielbachhuber was this change intentional? I'm treating the `.site-selector` as a block element instead of a flexible one because then the "Add new site" button would get close to the site list. This was the behavior we had before this PR, so I thought of maintaining it.

| Right now | How it was |
| ---------- | ------- |
| ![Screen Shot 2022-09-08 at 17 28 31](https://user-images.githubusercontent.com/26530524/189219851-ee373f45-ccd8-4a63-997a-4e619140daec.png) | ![Screen Shot 2022-09-08 at 17 29 39](https://user-images.githubusercontent.com/26530524/189219890-507cfa0f-76cb-41e7-93c9-42039028fb65.png) |

danielbachhuber replied:
@zaguiini Yep, it was: 83ec747f385c37b612a4785f076d664f396c94fe It was @p-jackson's preference.

p-jackson replied:
This project has a goal of not breaking Jetpack Cloud, and I think that's a balance. It doesn't mean _no_ changes at all to JP Cloud IMO.
Jetpack Cloud is a little different from regular Calypso, but I don't think it should needlessly diverge.
